lib/gyoku/xml_value.rb

```diff
@@ -16,22 +16,25 @@ class << self
 
       # Converts a given +object+ to an XML value.
       def create(object, escape_xml = true, options = {})
-        if Time === object
+        case object
+        when Time
           object.strftime XS_TIME_FORMAT
-        elsif DateTime === object
+        when DateTime
           object.strftime XS_DATETIME_FORMAT
-        elsif Date === object
+        when Date
           object.strftime XS_DATE_FORMAT
-        elsif String === object
+        when String
           escape_xml ? CGI.escapeHTML(object) : object
-        elsif object.respond_to?(:to_datetime)
-          create object.to_datetime
-        elsif object.respond_to?(:call)
+        when Proc
```

Comment:
Q: I wonder if my custom `#call`-supporting instance would match this `when`, or if the `object.respond_to?(:call)` needs to go down  into the `else`.

Reply:
the instance is not a Proc, so it we need to add it to the else block.

Reply:
Thanks for your care ❤️!